### PR TITLE
fixed BLI branch to be up to date with master

### PR DESCRIPTION
**Pull request by micah-bryant**
opening this so you can take a look if you want. I'll also add the python notebook I'm using

### CODE REVIEW

**Comment by aarmey** [comment]
Is it working or same problems as before?

File: src/BLI.jl [hunk 1]
```diff
@@ -9,3 +13,53 @@ end
...
+    plot(parse.(tps), parse.(bindData), title=cond, 
+         label=[conc[1] conc[2] conc[3] conc[4] conc[5] conc[6] conc[7] conc[8]], lw=3)
+end
+
+@model BLI(cond) = begin
+    Tshift = 822.2
+    Kon ~ Normal(6., 0.5)
+    Kdis ~ Normal(2., 1.)
+    
+    conc, tps, bindData = importData(cond)
```
Comment (aarmey) [inline src/BLI.jl]: @micah-bryant you should import the data outside of the `@model` object and then pass it in (like you do with `cond`). I think one of the things slowing everything down is every time the model is called it's reimporting the data.

File: src/BLI.jl [hunk 2]
```diff
@@ -9,3 +13,53 @@ end
...
+    Tshift = tps[1] + 600.01
+    theor_save = 0
+    exp_save = 0
+    for i in 1:length(conc)
+        tBind = tps[tps.<Tshift].-tps[1]
+        L0 = parse(Float64, conc[i])
```
Comment (aarmey) [inline src/BLI.jl]: Also don't prepare or parse the data here. Get it all tidied up outside of the model, then give the model a prepared array in exactly the structure it needs.

Comment (micah-bryant) [reply src/BLI.jl]: fixed it in the recent commit

File: src/BLI.jl (hunk 1 above)
Comment (micah-bryant) [reply src/BLI.jl]: fixed in the recent commit

**Comment by micah-bryant** [comment]
I tried the MvNormal distribution, but it was still taking a decent length of time. What I did instead was take the average of the residuals and define that to be a single normal distribution instead and the model built in ~1-2 min. Initial results are in the Notebook. Do these values look right? If not then I think that maybe the residuals aren't actually getting seen by the model and the values are being overwritten by the  ~Normal(u,std) line.

**Comment by micah-bryant** [comment]
I was wrong, the model currently builds in roughly 5 seconds

**Comment by micah-bryant** [comment]
I'm currently the lines of code below to try and make the model see the actual residuals, but am getting an error saying resStd < 0 which should be impossible according to the way Statistics.std() is calculated since it squares the difference. I'll keep you updated on what I find about this, but I'm doing this because looking at the results makes me think that the model isn't actually seeing the residuals since the means and std don't move from the initial guess.

resMean = mean(residuals)
resStd = std(residuals)
residual ~ Normal(resMean, resStd)

**Comment by aarmey** [comment]
What do you mean by doesn't move? You get 1000 iterations of the same #?

**Comment by aarmey** [comment]
You should compare each value individually to a Normal distribution, like you had before. The slow down was importing the data again on every iteration.

**Comment by micah-bryant** [comment]
At the end of 1000 iterations, the mean and standard deviation of the 3 distributions was within 0.03 of the starting points. I'll try the model again with the MvNormal.

**Comment by micah-bryant** [comment]
Just replaced with the MvNormal. It seems like it finishes all the iterations in a short amount of time (around 5-10 seconds).

**Pull request merged**